compiler/rustc_parse/src/lexer/tokentrees.rs

```diff
@@ -1,30 +1,12 @@
 use super::{StringReader, UnmatchedBrace};
-
 use rustc_ast::token::{self, Delimiter, Token};
 use rustc_ast::tokenstream::{DelimSpan, Spacing, TokenStream, TokenTree};
 use rustc_ast_pretty::pprust::token_to_string;
 use rustc_data_structures::fx::FxHashMap;
 use rustc_errors::{PErr, PResult};
 use rustc_span::Span;
 
-impl<'a> StringReader<'a> {
-    pub(super) fn into_token_trees(self) -> (PResult<'a, TokenStream>, Vec<UnmatchedBrace>) {
-        let mut tt_reader = TokenTreesReader {
-            string_reader: self,
-            token: Token::dummy(),
-            open_braces: Vec::new(),
-            unmatched_braces: Vec::new(),
-            matching_delim_spans: Vec::new(),
-            last_unclosed_found_span: None,
-            last_delim_empty_block_spans: FxHashMap::default(),
-            matching_block_spans: Vec::new(),
-        };
-        let res = tt_reader.parse_all_token_trees();
-        (res, tt_reader.unmatched_braces)
-    }
-}
-
-struct TokenTreesReader<'a> {
+pub(super) struct TokenTreesReader<'a> {
```

Comment:
No strong opinion, but keeping this think `private` seems beneficial

Reply:
Really, the module structure is backwards here. `TokenTreesReader` is the outermost layer, and should be in `lexer/mod.rs`, and then the lower-level `StringReader` should be in a module in `lexer/string_reader.rs`.

But instead we have `StringReader` in `lexer/mod.rs` and `TokenTreesReader` in `lexer/tokentrees.rs`. I decided that fixing this was too much for this PR. And exposing `TokenTreesReader` and a single method `parse_token_trees` via `pub(super)` seems pretty reasonable to me :)